tests/test_listen.py

```diff
@@ -0,0 +1,30 @@
+from unittest import mock
+
+from saluki.listen import listen
+
+
+def test_listen_with_partition_assigns_to_partition():
+    expected_partition = 123
+    topic = "sometopic"
+    with (
+        mock.patch(
+            "saluki.listen._deserialise_and_print_messages",
+            side_effect=KeyboardInterrupt,
+        ),
+        mock.patch("saluki.listen.Consumer") as c,
+    ):
+        listen("somebroker", "sometopic", partition=expected_partition)
+        assert c.return_value.assign.call_args[0][0][0].topic == topic
```

Comment:
Can we make `[0][0][0]` a little less obtuse? For example by naming some of the intermediates...